docs/reference/settings/security-settings.asciidoc

```diff
@@ -1947,6 +1947,311 @@ include::{es-repo-dir}/settings/common-defs.asciidoc[tag=ssl-supported-protocols
 include::{es-repo-dir}/settings/common-defs.asciidoc[tag=ssl-cipher-suites-values]
 // end::oidc-ssl-cipher-suites-tag[]
 
+[[ref-jwt-settings]]
+[discrete]
+===== JWT realm settings
+// tag::jwt-description-tag[]
+In addition to the <<ref-realm-settings,settings that are valid for all realms>>, you
+can specify the following settings.
+// end::jwt-description-tag[]
+
+// tag::jwt-allowed-issuer-tag[]
+`allowed_issuer` {ess-icon}::
+(<<static-cluster-setting,Static>>)
+A verifiable Identifier for your JWT Issuer. An Issuer
+Identifier is usually a case sensitive URL using the https scheme that contains
+scheme, host, and optionally, port number and path components and no query or
+fragment components. The value for this setting should be provided by your JWT
+Issuer.
+// end::jwt-allowed-issuer-tag[]
+
+// tag::jwt-allowed-signature-algorithm-tag[]
+`allowed_clock_skew` {ess-icon}::
+(<<static-cluster-setting,Static>>)
+The signature algorithm that will be used by {es} in order to verify the
+signature of the id tokens it will receive from the JWT Issuer.
+Defaults to `RSA256`.
+// end::jwt-allowed-signature-algorithm-tag[]
+
+// tag::jwt-allowed-clock-skew-tag[]
+`allowed_clock_skew` {ess-icon}::
+(<<static-cluster-setting,Static>>)
+The maximum allowed clock skew to be taken into consideration when validating
+id tokens with regards to their creation and expiration times.
+// end::jwt-allowed-clock-skew-tag[]
+
+// tag::jwt-jwkset-path-tag[]
+`jwkset_path` {ess-icon}::
+(<<static-cluster-setting,Static>>)
+The path or URL to a JSON Web Key Set with the key material that the JWT
+Issuer uses for signing tokens and claims responses.
+If a path is provided, then it is resolved relative to the {es} config
+directory.
+If a URL is provided, then it must be either a `file` URL or a `https` URL.
+{es} automatically caches the retrieved JWK set to avoid unnecessary HTTP
+requests but will attempt to refresh the JWK upon signature verification
+failure, as this might indicate that the JWT Provider has
+rotated the signing keys.
++
+File-based resources are polled at a frequency determined by the global {es}
+`resource.reload.interval.high` setting, which defaults to 5 seconds.
+// end::jwt-jwkset-path-tag[]
+
+// tag::jwt-allowed-audiences-tag[]
+`allowed_audiences` {ess-icon}::
+(<<static-cluster-setting,Static>>)
+A list of verifiable Identifiers for your JWT client applications.
+
+The OAuth 2.0 Client Identifier that was assigned to the
+OpenID Connect Relying Party application during registration
+at the OpenID Connect Provider.
```

Comment:
```suggestion
A list of allowed JWT audiences that {es} shoud verify.

{es} will only consume JWTs that were intended for one of
these audiences ( as denoted by the `aud` claim in the JWT )
```

We need to decouple this from RP client id, because while it might be this in many use cases, in the case of bespoke JWTs, there is no client_id and elasticsearch is the audience so we need to be a bit more generic here.